module/bdev/lvol/vbdev_lvol_rpc.c

```diff
@@ -1505,3 +1508,272 @@ rpc_bdev_lvol_shallow_copy_status(struct spdk_jsonrpc_request *request,
 
 SPDK_RPC_REGISTER("bdev_lvol_shallow_copy_status", rpc_bdev_lvol_shallow_copy_status,
 		  SPDK_RPC_RUNTIME)
+		  
+struct rpc_bdev_lvol_get_fragmap {
+	char *lvs_name;
+	char *lvol_name;
+	uint64_t offset;
+	uint64_t size;
+};
+
+struct fragmap_io {
+	struct spdk_bdev *bdev;
+	struct spdk_bdev_desc *bdev_desc;
+	struct spdk_io_channel *bdev_io_channel;
+	struct spdk_jsonrpc_request *request;
+
+	struct spdk_bit_array *fragmap;
+
+	uint64_t cluster_size;
+	uint64_t block_size;
+
+	uint64_t num_used_clusters;
+
+	uint64_t offset;
+	uint64_t size;
+	uint64_t current_offset;
+};
+
+static void
+free_rpc_bdev_lvol_get_fragmap(struct rpc_bdev_lvol_get_fragmap *r)
+{
+	free(r->lvs_name);
+	free(r->lvol_name);
+}
+
+static const struct spdk_json_object_decoder rpc_bdev_lvol_get_fragmap_decoders[] = {
+	{"lvol_name", offsetof(struct rpc_bdev_lvol_get_fragmap, lvol_name), spdk_json_decode_string, true},
+	{"lvs_name", offsetof(struct rpc_bdev_lvol_get_fragmap, lvs_name), spdk_json_decode_string, true},
+	{"offset", offsetof(struct rpc_bdev_lvol_get_fragmap, offset), spdk_json_decode_uint64, true},
+	{"size", offsetof(struct rpc_bdev_lvol_get_fragmap, size), spdk_json_decode_uint64, true},
+};
+
+static void seek_hole_done_cb(struct spdk_bdev_io *bdev_io, bool success, void *cb_arg);
+
+
+static char *
+bit_array_to_base64(const struct spdk_bit_array *ba)
```

Comment:
Why do you call this function `to_base_64`? It returns a string ... moreover, maybe it can be useful to move this function in `bitarray.c` file.

Reply:
It is used for encoding the bitmap to a base64 string. Do you have a better encoding method for the bitmap in the json response?

Reply:
No, this method is ok for me; what I meant is that a function that returns a string should have a name that ends with `_string`, something like `spdk_bit_array_to_base64_string`. But now with the description the purpose of the function is clearer.

Reply:
Okay, go your point. Thanks for the remindering of the convention.

Reply:
Updated. Thank you.